fluentd/generate_throttle_configs.rb

```diff
@@ -201,39 +217,37 @@ def get_project_pattern(name)
   project_pattern << "_*.log"
 end
 
-filename = "/etc/fluent/configs.d/user/throttle-conf.yaml"
-
-parsed = ""
-parsed = YAML.load_file(filename) if File.exists?(filename)
+filename = ENV['THROTTLE_CONF_LOCATION'] || DEFAULT_FILENAME
+parsed = if File.exists?(filename)
+             Hash[YAML.load_file(filename).map{|k,v|[k,v]}]
+         else
+            Hash.new
+         end 
 
 excluded = Array.new
 excludeSyslog = false
 
 # We do not yet support throttling logs read from the journal
 unless ENV['USE_JOURNAL'] == "true"
-  filename = "#{ENV['THROTTLE_CONF_LOCATION']}/settings"
-
-  parsed = ""
-  parsed = YAML.load_file(filename) if File.exists?(filename)
+  @log.info("Evaluating log trottle settings from #{filename}...")
 
-  parsed.each { |project|
-    name = project[0]
-    options = project[1]
+  parsed.each { |name, options|
+    @log.debug("Evaluating throttling for project '#{name}'")
 
     options.each_pair { |k,v|
 
       if validate(k,v)
         write_to_file(name, k, v, false)
         # build the list of paths the exclude
-        excluded.push(get_project_pattern(name)) unless name.eql?(".operations")
-
         if name.eql?(".operations")
-          excluded.push(get_project_pattern("default"))
-          excluded.push(get_project_pattern("openshift"))
-          excluded.push(get_project_pattern("openshift-infra"))
-
+          @log.debug("Found throttling settings for operations. Excluding projects: #{DEFAULT_OPS_PROJECTS}") 
+          DEFAULT_OPS_PROJECTS.each do |p|
+            excluded.push(get_project_pattern(p))
+          end
           excludeSyslog = true
           write_to_file(name, k, v, true)
+        else
+          excluded.push(get_project_pattern(name))
```

Comment:
It is but in a more traditional if/else construct.  Reading the original logic, I believe it still is valid.